File: app/src/main/java/com/dammaj/app/ui/AudioFragment.java
package com.dammaj.app.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.dammaj.app.R;
import com.dammaj.app.adapter.AudioAdapter;
import com.dammaj.app.data.ScraperCallback;
import com.dammaj.app.data.WebScraper;
import com.dammaj.app.model.AudioItem;

import java.util.ArrayList;
import java.util.List;

public class AudioFragment extends Fragment implements AudioAdapter.OnItemClickListener {

    private RecyclerView recyclerView;
    private ProgressBar progressBar;
    private View errorLayout;
    private TextView errorMessage;
    private Button retryButton;
    private AudioAdapter audioAdapter;
    private WebScraper webScraper;
    private List<AudioItem> audioList;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_audio, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        
        initViews(view);
        setupRecyclerView();
        setupWebScraper();
        loadAudioContent();
    }

    private void initViews(View view) {
        recyclerView = view.findViewById(R.id.recycler_view);
        progressBar = view.findViewById(R.id.progress_bar);
        errorLayout = view.findViewById(R.id.error_layout);
        errorMessage = view.findViewById(R.id.error_message);
        retryButton = view.findViewById(R.id.retry_button);
        
        audioList = new ArrayList<>();
        
        retryButton.setOnClickListener(v -> loadAudioContent());
    }

    private void setupRecyclerView() {
        audioAdapter = new AudioAdapter(audioList, this);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setAdapter(audioAdapter);
    }

    private void setupWebScraper() {
        webScraper = new WebScraper();
    }

    private void loadAudioContent() {
        showLoading();
        webScraper.scrapeAudio(new ScraperCallback() {
            @Override
            public void onSuccess(List<com.dammaj.app.model.ContentItem> contentItems) {
                // Not used in this fragment
            }

            @Override
            public void onFailure(Exception e) {
                if (getContext() != null) {
                    getContext().runOnUiThread(() -> {
                        hideLoading();
                        showError(e.getMessage() != null ? e.getMessage() : getString(R.string.error_loading));
                    });
                }
            }

            @Override
            public void onAudioSuccess(List<AudioItem> audioItems) {
                if (getContext() != null) {
                    getContext().runOnUiThread(() -> {
                        hideLoading();
                        audioList.clear();
                        audioList.addAll(audioItems);
                        audioAdapter.notifyDataSetChanged();
                    });
                }
            }

            @Override
            public void onAudioFailure(Exception e) {
                if (getContext() != null) {
                    getContext().runOnUiThread(() -> {
                        hideLoading();
                        showError(e.getMessage() != null ? e.getMessage() : getString(R.string.error_loading));
                    });
                }
            }
        });
    }

    private void showLoading() {
        progressBar.setVisibility(View.VISIBLE);
        recyclerView.setVisibility(View.GONE);
        errorLayout.setVisibility(View.GONE);
    }

    private void hideLoading() {
        progressBar.setVisibility(View.GONE);
        recyclerView.setVisibility(View.VISIBLE);
        errorLayout.setVisibility(View.GONE);
    }

    private void showError(String message) {
        progressBar.setVisibility(View.GONE);
        recyclerView.setVisibility(View.GONE);
        errorLayout.setVisibility(View.VISIBLE);
        errorMessage.setText(message);
    }

    @Override
    public void onItemClick(AudioItem item) {
        // Handle audio item click - play audio
        if (item.getAudioUrl() != null && !item.getAudioUrl().isEmpty()) {
            Toast.makeText(getContext(), "تشغيل: " + item.getTitle(), Toast.LENGTH_SHORT).show();
            
            // In a real implementation, you would:
            // 1. Initialize ExoPlayer
            // 2. Set the audio source
            // 3. Start playback
            // 4. Show playback controls
        }
    }

    @Override
    public void onPlayButtonClicked(AudioItem item) {
        // Handle play button click specifically
        onItemClick(item);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (webScraper != null) {
            webScraper.shutdown();
        }
    }
}

File: app/src/main/java/com/alilmia/app/ui/AudioFragment.kt
package com.alilmia.app.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.alilmia.app.R
import com.alilmia.app.adapter.AudioAdapter
import com.alilmia.app.data.ScraperCallback
import com.alilmia.app.data.WebScraper
import com.alilmia.app.databinding.FragmentAudioBinding
import com.alilmia.app.model.ContentItem
import com.alilmia.app.model.ContentType

/**
 * Audio fragment for audio content playback and management
 * Features list layout optimized for audio content with ExoPlayer integration
 */
class AudioFragment : Fragment() {

    private var _binding: FragmentAudioBinding? = null
    private val binding get() = _binding!!
    
    private lateinit var audioAdapter: AudioAdapter
    private lateinit var webScraper: WebScraper
    private var audioItems: MutableList<ContentItem> = mutableListOf()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentAudioBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        
        setupRecyclerView()
        setupWebScraper()
        
        // Load audio content
        loadAudioContent()
    }

    private fun setupRecyclerView() {
        audioAdapter = AudioAdapter { contentItem ->
            // Handle audio item click - play audio
            playAudio(contentItem)
        }
        
        binding.recyclerView.apply {
            layoutManager = LinearLayoutManager(context)
            adapter = audioAdapter
        }
    }

    private fun setupWebScraper() {
        webScraper = WebScraper()
    }

    private fun loadAudioContent() {
        showLoading(true)
        hideError()
        
        webScraper.scrapeHome(object : ScraperCallback {
            override fun onSuccess(items: List<ContentItem>) {
                activity?.runOnUiThread {
                    showLoading(false)
                    
                    // Filter only audio content
                    audioItems.clear()
                    audioItems.addAll(items.filter { it.type == ContentType.AUDIO })
                    audioAdapter.submitList(audioItems.toList())
                    
                    if (audioItems.isEmpty()) {
                        showError(getString(R.string.no_audio_content))
                    }
                }
            }

            override fun onFailure(error: Exception) {
                activity?.runOnUiThread {
                    showLoading(false)
                    showError(getString(R.string.error_loading_audio))
                    Toast.makeText(context, error.message, Toast.LENGTH_LONG).show()
                }
            }
        })
    }

    private fun playAudio(contentItem: ContentItem) {
        // TODO: Implement ExoPlayer audio playback
        // For now, show a toast with the audio title
        Toast.makeText(
            context,
            getString(R.string.playing_audio, contentItem.title),
            Toast.LENGTH_SHORT
        ).show()
        
        // Navigate to detail or show player UI
        navigateToDetail(contentItem)
    }

    private fun showLoading(show: Boolean) {
        binding.progressBar.visibility = if (show) View.VISIBLE else View.GONE
    }

    private fun showError(message: String) {
        binding.errorText.text = message
        binding.errorText.visibility = View.VISIBLE
        binding.recyclerView.visibility = View.GONE
    }

    private fun hideError() {
        binding.errorText.visibility = View.GONE
        binding.recyclerView.visibility = View.VISIBLE
    }

    private fun navigateToDetail(contentItem: ContentItem) {
        val intent = android.content.Intent(context, DetailActivity::class.java).apply {
            putExtra("content_item", contentItem)
        }
        startActivity(intent)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
        webScraper.shutdown()
    }

    companion object {
        fun newInstance() = AudioFragment()
    }
}